packages/next/src/client/components/router-reducer/reducers/server-patch-reducer.ts

```diff
@@ -14,22 +14,12 @@ export function serverPatchReducer(
   state: ReadonlyReducerState,
   action: ServerPatchAction
 ): ReducerState {
-  const { flightData, previousTree, overrideCanonicalUrl, cache, mutable } =
-    action
+  const { flightData, overrideCanonicalUrl, cache, mutable } = action
 
   const isForCurrentTree =
-    JSON.stringify(previousTree) === JSON.stringify(state.tree)
+    JSON.stringify(mutable.previousTree) === JSON.stringify(state.tree)
```

Comment:
We also still have to compare the `previousTree` that was passed into the action, because the tree is related to the data fetch you can't only assume the mutable is enough, it also needs to check if the tree at the time of the data fetch is still the same, and otherwise discard it.

Maybe that is not needed though and we can still apply the data 🤔